Mathlib/NumberTheory/LSeries/Basic.lean

```diff
@@ -12,140 +12,216 @@ import Mathlib.Analysis.NormedSpace.FiniteDimension
+lemma LSeriesSummable.LSeriesHasSum {f : ℕ → ℂ} {s : ℂ} (h : LSeriesSummable f s) :
     LSeriesHasSum f s (LSeries f s) :=
   h.hasSum
 
-lemma norm_LSeriesTerm_eq (f : ArithmeticFunction ℂ) (s : ℂ) (n : ℕ) :
-    ‖f n / n ^ s‖ = ‖f n‖ / n ^ s.re := by
-  rcases n.eq_zero_or_pos with rfl | hn
-  · simp only [map_zero, zero_div, norm_zero, zero_mul]
-  rw [norm_div, norm_natCast_cpow_of_pos hn]
+lemma LSeriesHasSum_iff {f : ℕ → ℂ} {s a : ℂ} :
+    LSeriesHasSum f s a ↔ LSeriesSummable f s ∧ LSeries f s = a :=
+  ⟨fun H ↦ ⟨H.LSeriesSummable, H.LSeries_eq⟩, fun ⟨H₁, H₂⟩ ↦ H₂ ▸ H₁.LSeriesHasSum⟩
 
-lemma norm_LSeriesTerm_le_of_re_le_re (f : ArithmeticFunction ℂ) {w : ℂ} {z : ℂ}
-    (h : w.re ≤ z.re) (n : ℕ) : ‖f n / n ^ z‖ ≤ ‖f n / n ^ w‖ := by
-  rcases n.eq_zero_or_pos with rfl | hn
-  · simp only [map_zero, CharP.cast_eq_zero, zero_div, norm_zero, le_refl]
-  have hn' := norm_natCast_cpow_pos_of_pos hn w
-  simp_rw [norm_div]
-  suffices H : ‖(n : ℂ) ^ w‖ ≤ ‖(n : ℂ) ^ z‖ from div_le_div (norm_nonneg _) le_rfl hn' H
-  refine (one_le_div hn').mp ?_
-  rw [← norm_div, ← cpow_sub _ _ <| cast_ne_zero.mpr hn.ne', norm_natCast_cpow_of_pos hn]
-  exact Real.one_le_rpow (one_le_cast.mpr hn) <| by simp only [sub_re, sub_nonneg, h]
-
-lemma LSeriesSummable.of_re_le_re {f : ArithmeticFunction ℂ} {w : ℂ} {z : ℂ} (h : w.re ≤ z.re)
-    (hf : LSeriesSummable f w) : LSeriesSummable f z := by
+lemma LSeriesHasSum_congr {f g : ℕ → ℂ} (s a : ℂ) (h : ∀ n ≠ 0, f n = g n) :
+    LSeriesHasSum f s a ↔ LSeriesHasSum g s a := by
+  simp only [LSeriesHasSum_iff, LSeriesSummable_congr s h, LSeries_congr s h]
+
+lemma LSeriesSummable.of_re_le_re {f : ℕ → ℂ} {s s' : ℂ} (h : s.re ≤ s'.re)
+    (hf : LSeriesSummable f s) : LSeriesSummable f s' := by
   rw [LSeriesSummable, ← summable_norm_iff] at hf ⊢
-  exact hf.of_nonneg_of_le (fun _ ↦ norm_nonneg _) (norm_LSeriesTerm_le_of_re_le_re f h)
+  exact hf.of_nonneg_of_le (fun _ ↦ norm_nonneg _) (norm_term_le_of_re_le_re f h)
 
-theorem LSeriesSummable_iff_of_re_eq_re {f : ArithmeticFunction ℂ} {w z : ℂ} (h : w.re = z.re) :
-    f.LSeriesSummable w ↔ f.LSeriesSummable z :=
+theorem LSeriesSummable_iff_of_re_eq_re {f : ℕ → ℂ} {s s' : ℂ} (h : s.re = s'.re) :
+    LSeriesSummable f s ↔ LSeriesSummable f s' :=
   ⟨fun H ↦  H.of_re_le_re h.le, fun H ↦ H.of_re_le_re h.symm.le⟩
-#align nat.arithmetic_function.l_series_summable_iff_of_re_eq_re ArithmeticFunction.LSeriesSummable_iff_of_re_eq_re
+#align nat.arithmetic_function.l_series_summable_iff_of_re_eq_re LSeriesSummable_iff_of_re_eq_re
+
 
-lemma LSeriesSummable.le_const_mul_rpow {f : ArithmeticFunction ℂ} {s : ℂ}
-    (h : LSeriesSummable f s) : ∃ C, ∀ n, ‖f n‖ ≤ C * n ^ s.re := by
+/-!
+### Criteria for and consequences of summability of L-series
+
+We relate summability of L-series with bounds on the coefficients in terms of powers of `n`.
+-/
+
+/-- If the `LSeries` of `f` is summable at `s`, then `f n` is bounded in absolute value
```

Comment:
The proof gives an upper bound that has some uniformity in `s`

Reply:
The statement is for a fixed `s`. Of course, we obtain the same bound for all `s'` with `re s' ≥ re s` from this.